### PR TITLE
Cache bust manifest request when the response is > 24 hours old.

### LINKED ISSUE #18: Impose max cache age of 24 hours for manifest requests
**Library Affected**:
sw-appcache-behavior

**Browser & Platform**:
all browsers

**Issue or Feature Request Description**:
There's an open [`// TODO: Consider cache-busting if the manifest response > 24 hours old.`](https://github.com/GoogleChrome/sw-helpers/blob/master/projects/sw-appcache-behavior/src/client-runtime.js#L139) that needs to be implemented, in order to protect against AppCache manifests that dangerously cache themselves indefinitely.

### PR DESCRIPTION
**Pull request by jeffposnick**
R: @jakearchibald @addyosmani @wibblymat @gauntface

Fixes #18 

This code checks the age of the response from the AppCache manifest request, and if it's older than 24 hours, makes another request using `cache: reload` to (ideally) bypass the cache, to work around http://alistapart.com/article/application-cache-is-a-douchebag#section6

I'm using `cache: reload` here, which unfortunately isn't universally supported in browsers right now. The alternative would be to append cache-busting URL parameters to the request, but I feel like it's in keeping with the spirit of this library to use the primitives provided by the platform. This is also an additional nice-to-have feature beyond the core functionality of the library, so if `cache: reload` doesn't have an effect, we're no worse off than we were.

### CODE REVIEW

**Comment by jeffposnick** [comment]
(I'm working with @gauntface to get some of the CI testing stuff cleaned up, and can hold off on merging this PR until that's done. Feel free to look at the code change in the meantime, though.)

**Comment by jeffposnick** [comment]
The tests are passing in all browsers with some recent changes merged in. I'd be grateful for any feedback on the code change from anyone on the R: line.

File: projects/sw-appcache-behavior/src/client-runtime.js [hunk 1]
```diff
@@ -136,8 +136,31 @@ function checkManifestVersion(db, manifestUrl) {
...
-    // TODO: Consider cache-busting if the manifest response > 24 hours old.
     fetch(manifestRequest).then(function(manifestResponse) {
+      var dateHeaderValue = manifestResponse.headers.get('date');
+      if (dateHeaderValue) {
+        var manifestDate = new Date(dateHeaderValue).valueOf();
+        // Calculate the age of the manifest in milliseconds.
+        var manifestAgeInMillis = Date.now() - manifestDate;
+        // If the age is greater than 24 hours, then we need to refetch without
+        // hitting the cache.
+        console.log(manifestAgeInMillis);
```
Comment (wibblymat) [inline projects/sw-appcache-behavior/src/client-runtime.js]: Remove debug line

**Comment by jeffposnick** [comment]
All good with the `console.log()` removed, @wibblymat?

**Comment by wibblymat** [comment]
Sorry, yes, also meant to say LGTM :)

**Pull request merged**